daml-model/daml/Trigger/LoanManagement.daml

```diff
@@ -0,0 +1,53 @@
+--
+-- Copyright (c) 2022, Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+-- SPDX-License-Identifier: Apache-2.0
+--
+
+module Trigger.LoanManagement where
+
+import DA.Action (void)
+import DA.Foldable (forA_)
+import DA.Optional
+import Daml.Trigger
+import Model.Balance
+import Model.LoanConfig
+
+
+loanManagementTrigger : Trigger ()
+loanManagementTrigger = Trigger
+    { initialize = pure ()
+    , updateState = \_ -> pure ()
+    , rule = createLoanWhenBelowThreshold
+    , registeredTemplates = RegisteredTemplates [
+        registeredTemplate @Balance,
+        registeredTemplate @LoanConfig
+        ]
+    , heartbeat = None
+    }
+
+createLoanWhenBelowThreshold : Party -> TriggerA () ()
+createLoanWhenBelowThreshold sender = do
+    loanConfigs <- query @LoanConfig
+    balances <- fmap (\x -> snd <$> x) $ query @Balance
+    forA_ loanConfigs \(loanCid, loanConfig) -> do
+        let maybeBalance = listToOptional $ filter (correspondsTo loanConfig) balances
+        case maybeBalance of
+            Some balance | loanConfig.activeLoan < loanConfig.maxLoanThreshold -> do
+                let amount = min loanConfig.maxLoanPerTransaction (loanConfig.maxLoanThreshold - loanConfig.activeLoan)
+                createLoan balance.provider balance.iban (Some amount) loanCid
+            Some balance | loanConfig.activeLoan >= loanConfig.maxLoanThreshold -> do
+                traceA $ "Maximum loan reached for " <> loanConfig.account
```

Comment:
If I am not wrong, this branch will be triggered again and again when the active loan is greater or equal than the threshold. (Not saying this is wrong, though.)

Reply:
@gaborh-da what would be the alternative? Because after a while, you might have payed back some of the loan, making the active loan smaller than the maxLoanThreshold.